src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java

```diff
@@ -716,6 +716,13 @@ public void settingsV2(
     public void startIndex(
         StartIndexRequest startIndexRequest, StreamObserver<StartIndexResponse> responseObserver) {
       logger.info("Received start index request: {}", startIndexRequest);
+      if (startIndexRequest.getIndexName().isEmpty()) {
+        responseObserver.onError(
```

Comment:
You can add a logger.warn() before this error handling to add info on the indexName for which this failed. And make the error more descriptive such as "error while trying to start index since indexName was empty: " + indexName

Reply:
Yup, added warning with descriptive message

Reply:
Empty index name in an error might be confusing (and it's always going to be ""). Maybe skip the index name or log the start index request?

Reply:
Logging the startIndex request sounds like a better option. Will it show when the indexName string is not set vs an empty string since it seems like protobuf sets string defaults to ""?

Reply:
It doesn't show unset only empty unfortunately

Reply:
Thank you for reviewing this! @sarthakn7 @swekannan 
Since line 718 has already logged out the `startIndexRequest`, here I skipped the index name and warned it's an empty index name.